### DISCUSSION TITLE
Modal Dialog/Prompt Appears in Main Screen Instead of Popout Window

### DISCUSSION THREAD

**Discussion by avinash-bsy** [iTwin/appui#938] [Q&A] [answered]
I’ve been encountering an issue with modal dialogs and prompts in popout windows. Whenever I’m in a popout mode and I click to open a modal dialog or prompt, it appears in the main screen rather than the popout window.

This behavior is quite disruptive as it requires me to switch back and forth between the main screen and the popout window. I was wondering if anyone else has experienced this issue and if there are any known solutions or workarounds.

Here’s a brief description of the issue:

Open a popout window.
Click to open a modal dialog or prompt within the popout window.
The modal dialog or prompt appears in the main screen instead of the popout window.
Any help or guidance would be greatly appreciated. Thank you!

**Comment by GerardasB** [comment, marked as answer]
Hey, what APIs exactly are you using, could you try using the [iTwinUI Dialog](https://itwinui.bentley.com/docs/dialog) (make sure `portal` prop is set to `true` or use the `Modal` dialog), it should automatically append itself to the correct document based on the `portalContainer` of a wrapping `ThemeProvider`.
Alternatively, if you are using [UiFramework.dialogs.modal.open()](https://www.itwinjs.org/reference/appui-react/dialog/frameworkdialog/) it takes a `parentDocument?: Document` as a third argument. You could get a ref to one of the HTML elements in your widget and use `ownerDocument` to specify that arg.

**Reply by avinash-bsy** [reply]
We are using the later method, UiFramework.dialogs.modal.open method. I tried passing parentDocument parameter as you mentioned

parentDocument = IModelApp.viewManager.selectedView?.vpDiv?.ownerDocument

But the issue continues, Dialog still comes up in the first screen instead of the popped out window. It would of great help if you could elaborate on this.

"it takes a parentDocument?: Document as a third argument. You could get a ref to one of the HTML elements in your widget and use ownerDocument to specify that arg."

**Reply by GerardasB** [reply]
This PR adds an example with the described approach: https://github.com/iTwin/appui/pull/1209

https://github.com/iTwin/appui/blob/7f5d60fe6464a88508116ac57db0fa5c0d5de48a/apps/test-app/src/frontend/appui/frontstages/ElementStacking.tsx#L219
